### ISSUE TITLE
Detecting ADS being closed in order to show a warning popup before closing the window

### ISSUE THREAD

**Issue by goyal-anjali** [microsoft/azuredatastudio#23239]
With respect to Database Migration Assessment for Oracle, when an assessment is in progress and customer tries to close ADS, we want to give them a popup.
Something like - 'Assessment is in progress. Do you want to close Azure Data Studio?'

**Question**
How can we detect ADS close button getting clicked? Is there an event to track that?

**Comment by alanrenmsft** [comment]
@aasimkhan30 let me know if you need any help.

**Comment by Charles-Gagnon** [comment]
You could use a task for this - it will display a prompt when the user quits if it's still running asking if they want to quit.

https://github.com/microsoft/azuredatastudio/blob/5560b52df1651501623589e0ced226f083e57ab0/src/sql/workbench/services/tasks/common/tasksService.ts#L152

**Comment by goyal-anjali** [comment]
Our extension repo sits outside ADS, can I still use it? @Charles-Gagnon

**Comment by Charles-Gagnon** [comment]
Yes, tasks are part of the extensibility API (the code I linked was just showing where in code it does the popup. That isn't something you control directly except by having an actively running task)

See https://github.com/microsoft/azuredatastudio/blob/0fca085ca7c3f2b329d078f2222dd7b1cca387b1/extensions/sql-database-projects/src/models/deploy/deployService.ts#L138 for an example of starting a task
